### PR TITLE
DashScope: Support tool-call for QwenStreamingChatModel

### PR DESCRIPTION
**Pull request by jiangsier-xyz**
## Issue

Closes #1994

## Change
Support tool-call for QwenStreamingChatModel.
And Optimize switch-case statements.

## General checklist

- [X] There are no breaking changes
- [X] I have added unit and integration tests for my change
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

### CODE REVIEW

File: langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenHelper.java [hunk 1]
```diff
@@ -388,8 +370,8 @@ static ToolBase toToolFunction(ToolSpecification toolSpecification) {
     private static JsonObject toParameters(ToolSpecification toolSpecification) {
         if (toolSpecification.parameters() != null) {
             return JsonUtils.toJsonObject(toMap(toolSpecification.parameters()));
-        } else if (toolSpecification.toolParameters() != null) {
-            return JsonUtils.toJsonObject(toolSpecification.toolParameters());
+        } else if (toolSpecification.parameters() != null) {
+            return JsonUtils.toJsonObject(toolSpecification.parameters());
```
Comment (langchain4j) [inline langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenHelper.java]: This does not seem right, `toolSpecification.parameters()` is already handled above.

We should keep `toolSpecification.parameters()` support for backwards-compatibility

File: langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenChatModelIT.java [hunk 2]
```diff
@@ -99,7 +94,9 @@ public void should_call_function_with_argument_then_answer(String modelName) {
         ToolSpecification hasArgToolSpec = ToolSpecification.builder()
                 .name(toolName)
                 .description("Query the weather of a specified city")
-                .addParameter("cityName", JsonSchemaProperty.STRING)
+                .parameters(JsonObjectSchema.builder()
```
Comment (langchain4j) [inline langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenChatModelIT.java]: I would keep this IT with an old API to make sure it works for users that still use it

File: langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenStreamingChatModelIT.java [hunk 3]
```diff
@@ -40,6 +44,192 @@ public void should_send_non_multimodal_messages_and_receive_response(String mode
         assertThat(response.content().text()).endsWith("That's all!");
     }
 
+    @ParameterizedTest
+    @MethodSource("dev.langchain4j.model.dashscope.QwenTestHelper#functionCallChatModelNameProvider")
```
Comment (langchain4j) [inline langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenStreamingChatModelIT.java:49]: Could you please also check `StreamingChatLanguageModelIT` and `StreamingAiServicesSimpleIT` from https://github.com/langchain4j/langchain4j/pull/1989 and make sure `QwenStreamingChatModel` passes them? You can just copy those tests and run them locally, no need to include them in this PR. Thanks!

**Review by ghost — COMMENTED** [review]
@jiangsier-xyz thank you!

File: langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenChatModelIT.java (hunk 2 above)
Comment (jiangsier-xyz) [reply langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenChatModelIT.java]: > I would keep this IT with an old API to make sure it works for users that still use it

Are you suggesting that I should continue using addParameter("cityName", JsonSchemaProperty.STRING) in the test code?

Comment (langchain4j) [reply langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenChatModelIT.java]: Yes, maybe not in all tests, but al least in a few ones. `addParameter()` (old API) and `parameters(JsonObjectSchema)` (new API) should both work untill 1.0 version. Please note that `QwenAiServicesWithToolsIT` already tests new API.

File: langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenHelper.java (hunk 1 above)
Comment (jiangsier-xyz) [reply langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenHelper.java]: Rolled back.

File: langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenChatModelIT.java (hunk 2 above)
Comment (jiangsier-xyz) [reply langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenChatModelIT.java]: Rolled back.

File: langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenStreamingChatModelIT.java (hunk 3 above)
Comment (jiangsier-xyz) [reply langchain4j-dashscope/src/test/java/dev/langchain4j/model/dashscope/QwenStreamingChatModelIT.java:49]: > Could you please also check `StreamingChatLanguageModelIT` and `StreamingAiServicesSimpleIT` from #1989 and make sure `QwenStreamingChatModel` passes them? You can just copy those tests and run them locally, no need to include them in this PR. Thanks!

Yes, the QwenStreamingChatModel can pass the 2 tests.

**Review by ghost — APPROVED** [review]
@jiangsier-xyz thank you!

**Pull request merged**